activiti-core/activiti-engine/src/main/java/org/activiti/engine/delegate/event/impl/ActivitiEntityEventImpl.java

```diff
@@ -24,18 +24,30 @@
  */
 public class ActivitiEntityEventImpl extends ActivitiEventImpl implements ActivitiEntityEvent {
 
-  protected Object entity;
+    protected Object entity;
 
-  public ActivitiEntityEventImpl(Object entity, ActivitiEventType type) {
-    super(type);
-    if (entity == null) {
-      throw new ActivitiIllegalArgumentException("Entity cannot be null.");
+    protected String reason;
+
+    public ActivitiEntityEventImpl(Object entity,
+                                   ActivitiEventType type) {
+        super(type);
+        if (entity == null) {
+            throw new ActivitiIllegalArgumentException("Entity cannot be null.");
+        }
+        this.entity = entity;
+    }
+
+    @Override
+    public Object getEntity() {
+        return entity;
     }
-    this.entity = entity;
-  }
 
-  @Override
-  public Object getEntity() {
-    return entity;
-  }
+    @Override
+    public String getReason() {
+        return reason;
+    }
+
+    public void setReason(String reason) {
```

Comment:
is this setter really necessary? can thew reason be passed as constructor param?

Reply:
Also, commited by mistake and directly related to the above comment